crates/astria-sequencer/src/app/mod.rs

```diff
@@ -1024,10 +1025,10 @@ impl App {
 
         // flag mempool for cleaning if we ran a fee change action
         self.recost_mempool = self.recost_mempool
-            || signed_tx
-                .actions()
-                .iter()
-                .any(|action| matches!(action, Action::FeeAssetChange(_) | Action::FeeChange(_)));
+            || matches!(signed_tx.group(), Some(ActionGroup::BundlableSudo(_)))
+                && signed_tx.actions().iter().any(|action| {
+                    matches!(action, Action::FeeAssetChange(_) | Action::FeeChange(_))
```

Comment:
It's idiomatic to implement these as inherent methods on an enum so that this would read like:

```rust
// inherent method
impl Action {
    fn is_fee_asset_change(&self) -> bool {
        matches!(action, Self::FeeAssetChange(_))
    }
    fn is_fee_change(&self) -> bool {
        matches!(action, Self::FeeChange(_))
    }
}

// the function
&& signed_tx.actions().iter().any(|act| act.is_fee_asset_change() || act.is_fee_change())
```